### PR TITLE
Configuration options and get/set methods

### PR DESCRIPTION
**Pull request by andres-urquieta**

### CODE REVIEW

File: src/api/index.js [hunk 1]
```diff
@@ -30,11 +30,14 @@ async function loadSitemap(resource) {
   await requestResource({ href, method: 'get', resourceKey: 'sitemap' })
 }
 
-async function processRequest({ data, href, id, method, resourceKey, onDone }) {
+async function processRequest({ data, href, id, method, resourceKey,
+  name, config }) {
```
Comment (nranas) [inline src/api/index.js]: We should really change the "Maximum row length" setting..

Comment (hfjallemark) [inline src/api/index.js]: +1

File: themes/default/document.css [hunk 2]
```diff
@@ -5,14 +5,25 @@
...
-  border-bottom: 1px solid #eee;
   display: flex;
   margin-bottom: 30px;
   padding: 10px 0 20px;
 }
 
+.ct-document-header {
+  border-bottom: 1px solid #eee;
+  margin-bottom: 30px;
+  padding: 10px 0 20px;
```
Comment (nranas) [inline themes/default/document.css:16]: These two rows is duplicated in both rules. They only need to be in one of them, right?

File: README.md [hunk 3]
```diff
@@ -86,8 +95,10 @@ Properties should be provided in camel-casing **without** the data-prefix, e.g.
 | Attribute  | Configuration property | Value type | Description |
 | ---------- | ---------------------- | ---------- | ----------- | 
 | `data-endpoint` | `endpoint` | string | Specifies the starting point of the API that should be rendered. |
+| `data-action-list-position` | `actionListPosition` | string | Specifies the position of the action buttons. Accepted values are: `top` (default), `bottom` and `both`. |
 | | `onValueChange` | function | Callback function executed after a value has been updated. The callback receives 2 parameters: `id` (of the updated element) and `value` (after the change). |
 | | `onUrlChange` | function | Callback function executed after the url has changed. Receives 1 parameter: `url` (after the navigation). |
+| | `onProcessComplete` | function | Callback function executed when the process has ended. Receives 2 parameters: `url` (where the client would normally redirect) and `content` (of that url). If the function exists, the process will NOT automatically redirect. Optional return value: `{ title, content }`. |
```
Comment (hfjallemark) [inline README.md]: The hypermedia api doesn't really know anything about processes, can we rename this to something more appropriate that is not specific to LM?

File: src/components/document.js [hunk 4]
```diff
@@ -9,10 +9,12 @@ function getHref(resource) {
 
 export default class Document extends Component {
   componentDidUpdate(prevProps, prevState) {
-    const prev = getHref(prevProps.resource)
-    const curr = getHref(this.props.resource)
-    if (prev !== curr) {
-      this.props.config.onUrlChange(curr)
+    if (this.props.config.onUrlChange) {
+      const prev = getHref(prevProps.resource)
```
Comment (hfjallemark) [inline src/components/document.js]: Can we use full names?

File: src/components/document.js [hunk 5]
```diff
@@ -9,10 +9,12 @@ function getHref(resource) {
...
+      const curr = getHref(this.props.resource)
```
Comment (hfjallemark) [inline src/components/document.js]: Same here.

**Pull request merged**